DatadogSessionReplay/Sources/Writer/Models/EnrichedResource.swift

```diff
@@ -0,0 +1,42 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+#if os(iOS)
+import Foundation
+
+/// Extends the resource information with context.
+internal struct EnrichedResource: Codable, Resource {
```

Comment:
Do you need to serialize and deserialize both?

If not, let's not codegen unnecessary for both.

Reply:
Yeah - it's serialized to a batch when writing, and deserialized before sending.

Reply:
We must not deserialize before sending, for sending the data written to disk, we should consider them raw bytes like we do any other events.

Although, I agree we deserialize for filtering but that's just the metadata, not the real payload.

Reply:
I'm afraid we have no other way to store images using current api of the core, we need to serialize the image in a json and deserialise it before upload. Which mean doing base64 encoding/decoding for each one of them.
@maciejburda did you take a look at cpu overhead?

Reply:
Yeah - for session replay it's been different since the beginning because we rely on the multipart form in a certain way.

It's on our radar to change it, but it's actually not trivial.